server/contexts/logging/logging.go

```diff
@@ -105,8 +105,17 @@ func (l *LoggingContext) Log(ctx context.Context, logger kitlog.Logger) {
 		"took", time.Since(l.StartTime),
 	)
 
-	if l.Err != nil {
-		keyvals = append(keyvals, "err", l.Err)
+	if l.hasErr() {
+		for _, err := range l.Errs {
+			errKey := "err"
```

Comment:
Using the same key might be an issue if folks configure the logging to be JSON. I wonder if we need to actually hold the whole chain of errors or just the most recent one that was attached to the context?

Reply:
We could just log the slice, which would work with JSON.

Another idea was a map[string]error, or basically more keyvals, so that you can add a key for the errors, but it's not consistent. Repeated keys can already be an issue though.

Reply:
I'm open to whichever approach you like.